sklearn/mixture/gmm.py

```diff
@@ -411,7 +411,30 @@ def sample(self, n_samples=1, random_state=None):
                     num_comp_in_X, random_state=random_state).T
         return X
 
-    def fit(self, X, y=None):
+    def fit_predict(self, X, y=None):
+        """
+        Fit and then predict labels for data.
+        Warning: due to the final maximization step in the EM algorithm,
+        with low iterations the prediction may not be 100% accurate
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+
+        Returns
+        -------
+        C : array, shape = (n_samples,)
+        """
+
+        responsibilities = self._fit(X, y)
+        if responsibilities is None:
+            prediction = None
```

Comment:
It could happen in the case when n_iter == 0. You're right that never returning None is better, so I added a check to run score_samples to get the correct output value in that case (that apparently happens when running an HMM).